### PR TITLE
PUIS-79

### PR DESCRIPTION
**Pull request by arpargo**
Changed:
Docker base image
Upgraded node version to 18
Removed node-sass library and it's extensions.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**PUISTOTALKOOT-UI branch is deployed to platta: https://puistotalkoot-pr187-ui.dev.hel.ninja** :rocket::rocket::rocket:

File: package.json [hunk 1]
```diff
@@ -2,6 +2,9 @@
   "name": "linked-volunteering-ui",
   "version": "0.2.7",
   "private": true,
+  "engines": {
+    "node": ">=16.0.0 <18.0.0"
```
Comment (mikkojamG) [inline package.json:6]: Could also add `.nvmrc` file 🤔

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,12 @@
-# Our First stage, that builds the application
-FROM helsinkitest/node:12-slim AS react-builder
+# ==========================================
+FROM registry.access.redhat.com/ubi8/nodejs-16 AS deployable
```
Comment (mikkojamG) [inline Dockerfile:2]: Should try to favour at least Node.js 18 if possible, since it's currently oldest LTS version.

File: package.json [hunk 3]
```diff
@@ -94,19 +93,19 @@
...
     "husky": "^4.2.3",
     "prettier": "^1.19.1",
+    "sass": "^1.75.0",
     "serve": "^11.3.0"
   },
   "scripts": {
-    "start": "yarn setPackageDetails react-scripts start",
-    "build": "yarn setPackageDetails react-scripts build",
+    "start": "yarn react-scripts start",
+    "build": "yarn react-scripts build",
```
Comment (mikkojamG) [inline package.json:101]: I think having yarn here is a bit redundant 👀

File: src/config/theme.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable */
-require('./bootstrap.scss');
-const extractedTheme = require('sass-extract-loader?{"plugins":["sass-extract-js"]}!./theme.scss');
-/* eslint-enable */
+const theme = require('./bootstrap.scss');
+
+console.log(theme);
```
Comment (mikkojamG) [inline src/config/theme.js:4]: Console.log could be removed.

**Review by mikkojamG — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (arpargo) [reply Dockerfile:2]: Very valid point but I struggled for half a day with upgrading to 18 without any meaningful progress and followed the instruction of minimal work when downgrading to 16 fixed all the issues that are coming on upgrade to 18.

Comment (charn) [reply Dockerfile:2]: Could be ubi9 instead of ubi8

Comment (arpargo) [reply Dockerfile:2]: ubi9 version of the image fails to build
Error: error:0308010C:digital envelope routines::unsupported       

As this was asked to be least effort I'll not resolve this right now.

**Comment by terovirtanen** [comment]
**PUISTOTALKOOT-UI branch is deployed to platta: https://puistotalkoot-pr187-ui.dev.hel.ninja** :rocket::rocket::rocket:

File: .nvmrc [hunk 5]
```diff
@@ -0,0 +1 @@
+16
```
Comment (mikkojamG) [inline .nvmrc:1]: 16.20.2 would match the one in Docker image 👀

**Comment by terovirtanen** [comment]
**PUISTOTALKOOT-UI branch is deployed to platta: https://puistotalkoot-pr187-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**PUISTOTALKOOT-UI branch is deployed to platta: https://puistotalkoot-pr187-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Review by mikkojamG — APPROVED** [review]
LGTM 👀

**Comment by terovirtanen** [comment]
**PUISTOTALKOOT-UI branch is deployed to platta: https://puistotalkoot-pr187-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**PUISTOTALKOOT-UI branch is deployed to platta: https://puistotalkoot-pr187-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Pull request merged**